fem/bilininteg_diffusion.cpp

```diff
@@ -623,7 +623,7 @@ void PADiffusionApply3D(const int NE,
       const int Q1D = T_Q1D ? T_Q1D : q1d;
       constexpr int max_D1D = T_D1D ? T_D1D : MAX_D1D;
       constexpr int max_Q1D = T_Q1D ? T_Q1D : MAX_Q1D;
-      double grad[max_Q1D][max_Q1D][max_Q1D][4];
+      double grad[max_Q1D][max_Q1D][max_Q1D][3];
```

Comment:
I think I've seen this `4` before -- I wasn't sure if it's really necessary -- sometimes the extra index is used to store the interpolated value in addition to the 3 derivatives. Looking at the code now, it looks like the last index is never used, so `3` should be OK.

Reply:
Yes exactly. I'm also thinking that the memory might have been re-used to hold all 4 2nd derivatives, but then making the assumption that `xy == yx` simplified the memory footprint.

Reply:
This '4' comes from the `occa-dev` branch (https://github.com/mfem/mfem/blob/cf3bc930faf468f413fb7687897c39616d7361b2/kernels/fem/diffusion/tensor/cpu.okl#L228) : I think it was initially used as a '3+padding' to reach better performances but I haven't checked it since.

Reply:
Oh, yes, using `4` for padding makes sense. Should we keep it `4` then?